resources/type-system/inference.md

```diff
@@ -698,15 +770,32 @@ to `L` under constraints `C`.
   under constraints `Ci`.
 - Function types with named parameters are treated analogously to the positional
   parameter case above.
-- A generic function type `<T0 extends B0, ..., Tn extends Bn>F0` is a subtype
-match for a generic function type `<S0 extends B0, ..., Sn extends Bn>F1` with
-respect to `L` under constraints `Cl`:
-  - If `F0[Z0/T0, ..., Zn/Tn]` is a subtype match for `F0[Z0/S0, ..., Zn/Sn]`
-with respect to `L` under constraints `C`, where each `Zi` is a fresh type
-variable with bound `Bi`.
-  - And `Cl` is `C` with each constraint replaced with its closure with respect
+
+- A generic function type `<T0 extends B00, ..., Tn extends B0n>F0` is a subtype
+match for a generic function type `<S0 extends B10, ..., Sn extends B1n>F1` with
+respect to `L` under constraint set `C2`
+  - If `B0i` is a subtype match for `B1i` with constraint set `Ci0`
+  - And `B1i` is a subtype match for `B0i` with constraint set `Ci1`
```

Comment:
What are examples of non-equal `B0` and `B1` when `B0 <# B1` and `B1 <# B0`?
Only top types `dynamic`, `Object?`, and `void`?

Reply:
`void Function<T extends S>()` is a subtype match for `void Function<T extends int>()` with respect to `{S}` under constraint set `{int <: S <: int}`.